common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -328,6 +328,10 @@ public MergedBlockMeta getMergedBlockMeta(
     int size = (int) indexFile.length();
     // First entry is the zero offset
     int numChunks = (size / Long.BYTES) - 1;
+    if (numChunks <= 0) {
```

Comment:
From the stack-trace, this ArithmeticException would occur at `blockSize / bitmaps.length` when `bitmaps.length` is 0. Why not check the length of `chunkBitMaps` instead?

Reply:
`chunkBitMaps` is of type `FileSegmentManagedBuffer` and is deserialized on the client.